posthog/api/event_definition.py

```diff
@@ -27,4 +34,25 @@ class EventDefinitionViewSet(
     search_fields = ["name"]
 
     def get_queryset(self):
+        if self.request.user.organization.is_feature_available("event_property_collaboration"):  # type: ignore
+            from ee.models.event_definition import EnterpriseEventDefinition
+
+            return self.filter_queryset_by_parents_lookups(EnterpriseEventDefinition.objects.all()).order_by(
+                self.ordering
+            )
         return self.filter_queryset_by_parents_lookups(EventDefinition.objects.all()).order_by(self.ordering)
+
+    def get_serializer_class(self) -> Type[serializers.ModelSerializer]:
+        serializer_class = self.serializer_class
+        if self.request.user.organization.is_feature_available("event_property_collaboration"):  # type: ignore
+            from ee.api.enterprise_event_definition import EnterpriseEventDefinitionSerializer
+
+            serializer_class = EnterpriseEventDefinitionSerializer  # type: ignore
+        return serializer_class
+
+    def retrieve(self, request: Request, *args: Any, **kwargs: Any):
+        if self.request.user.organization.is_feature_available("event_property_collaboration"):  # type: ignore
```

Comment:
Tagging @timgl who probably has better context but I'm not sure if overriding each action in the viewset is the right way. I think I would probably have a standalone enterprise viewset and override the entire thing at the URL level.

Reply:
That was the original plan but we weren't able to call if the user was enterprise or not (wasn't able to call models in the routes setup) and there wasn't a setting for it, so this was the alternative. I'm happy to move this into separate viewsets if we can find a way to check that conditional in the urls or __init__ file

Reply:
That makes sense. In that case, just overriding the `get_serializer_class` should be enough. If you're keeping reliance on syncing the enterprise and regular table you'll need to override the `get_queryset` too, but if not just the serializer class should be enough.